wweval/R/combine_and_summarize_ww_data.R

```diff
@@ -0,0 +1,423 @@
+#' summary statistics for that forecast date and location
+#' @export
+load_data_and_summarize <- function(fp_hosp, fp_ww,
+                                    this_forecast_date,
+                                    this_location) {
+  # Use hospital data to get state pop
+  this_hosp_data <- readr::read_tsv(fp_hosp)
+  state_pop <- this_hosp_data |>
+    dplyr::distinct(.data$pop) |>
+    dplyr::pull()
+  if (length(state_pop) != 1) {
+    cli::cli_abort(message = "multiple state pops reported")
+  }
+
+  this_ww_data <- readr::read_tsv(fp_ww)
+  if (!nrow(this_ww_data) == 0) {
+    n_sites <- this_ww_data |>
+      dplyr::distinct(.data$site) |>
+      nrow()
+    n_labs <- this_ww_data |>
+      dplyr::distinct(.data$lab) |>
+      nrow()
+    sum_site_pops <- this_ww_data |>
+      dplyr::group_by(.data$site) |>
+      dplyr::summarise(
+        mean_site_pop = mean(.data$ww_pop, na.rm = TRUE)
+      ) |>
+      dplyr::ungroup() |>
+      dplyr::summarise(ww_total_pop = sum(.data$mean_site_pop,
+        na.rm = TRUE
+      )) |>
+      dplyr::pull(.data$ww_total_pop)
+    pop_coverage <- sum_site_pops / state_pop
+
+    avg_latency <- this_ww_data |>
+      dplyr::group_by(.data$lab_wwtp_unique_id) |>
+      dplyr::summarize(max_date = max(.data$date)) |>
+      dplyr::mutate(
+        latency = as.numeric(lubridate::ymd(
+          !!this_forecast_date
+        ) - lubridate::ymd(.data$max_date))
+      ) |>
+      dplyr::summarize(
+        mean_latency = mean(.data$latency, na.rm = TRUE)
+      ) |>
+      dplyr::pull(.data$mean_latency)
+
+    avg_sampling_freq <- this_ww_data |>
+      dplyr::group_by(.data$lab_wwtp_unique_id) |>
+      dplyr::arrange(.data$date, desc = TRUE) |>
+      # There are some duplicate dates within a site and lab
+      dplyr::distinct(.data$date) |>
+      dplyr::summarize(
+        n_days_w_samples = dplyr::n(),
+        n_days_total = as.numeric(max(.data$date) - min(.data$date)),
+        mean_collection_freq = .data$n_days_w_samples / .data$n_days_total
+      ) |>
+      dplyr::summarize(
+        mean_mean_collection_freq = mean(.data$mean_collection_freq)
+      ) |>
+      dplyr::pull(mean(.data$mean_mean_collection_freq))
+
+    n_days_w_samples <- this_ww_data |>
+      dplyr::summarize(n_days_w_samples = dplyr::n()) |>
+      dplyr::pull(.data$n_days_w_samples)
+
+    n_duplicate_obs <- this_ww_data |>
+      dplyr::group_by(.data$lab_wwtp_unique_id, .data$date) |>
+      dplyr::summarize(n_obs = dplyr::n()) |>
+      dplyr::ungroup() |>
+      dplyr::summarize(
+        n_duplicates = sum(.data$n_obs > 1)
+      ) |>
+      dplyr::pull(.data$n_duplicates)
+
+    this_ww_metadata <- tibble::tibble(
+      forecast_date = !!this_forecast_date,
+      location = !!this_location,
+      ww_data_present = 1,
+      n_sites,
+      n_labs,
+      pop_coverage,
+      state_pop,
+      avg_latency,
+      avg_sampling_freq,
+      n_days_w_samples,
+      n_duplicate_obs
```

Comment:
This large block could be a `summarize_ww_metadata` function?

Reply:
Not essential

Reply:
I think that is mostly what this is doing! Would rather not break into multiple functions